internal/cli/runner_install.go

```diff
@@ -0,0 +1,259 @@
+package cli
+
+import (
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/installutil"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	"github.com/hashicorp/waypoint/internal/runnerinstall"
+	"github.com/hashicorp/waypoint/pkg/server"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/hashicorp/waypoint/pkg/serverconfig"
+	"github.com/posener/complete"
+	"sort"
+	"strings"
+)
+
+type RunnerInstallCommand struct {
+	*baseCommand
+
+	platform            string `hcl:"platform,required"`
+	adopt               bool   `hcl:"adopt,optional"`
+	serverUrl           string `hcl:"server_url,required"`
+	serverCookie        string `hcl:"server_cookie,optional"`
+	id                  string `hcl:"id,optional"`
+	serverTls           bool   `hcl:"server_tls,optional"`
+	serverTlsSkipVerify bool   `hcl:"server_tls_skip_verify,optional"`
+	serverRequireAuth   bool   `hcl:"server_require_auth,optional"`
+}
+
+func (c *RunnerInstallCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *RunnerInstallCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *RunnerInstallCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:   "platform",
+			Usage:  "Platform to install the Waypoint runner into.",
+			Target: &c.platform,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "server-addr",
+			Usage:  "Address of the Waypoint server.",
+			EnvVar: "WAYPOINT_ADDR",
+			Target: &c.serverUrl,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "server-tls",
+			Target:  &c.serverTls,
+			Usage:   "If true, the Waypoint runner will connect to the server over TLS.",
+			Default: true,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "server-tls-skip-verify",
+			Target: &c.serverTlsSkipVerify,
+			Usage:  "If true, will not validate TLS cert presented by the server.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "server-require-auth",
+			Target: &c.serverRequireAuth,
+			Usage:  "If true, will send authentication details.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "adopt",
+			Usage:   "Adopt the runner after it is installed.",
+			Default: true,
+			Target:  &c.adopt,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "server-cookie",
+			Usage:  "Server cookie for the Waypoint cluster for which you're targeting this runner.",
+			Target: &c.serverCookie,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "id",
+			Usage:  "If this is set, the runner will use the specified id.",
+			Target: &c.id,
+		})
+
+		// Add platforms in alphabetical order. A consistent order is important for repeatable doc generation.
+		i := 0
+		sortedPlatformNames := make([]string, len(runnerinstall.Platforms))
+		for name := range runnerinstall.Platforms {
```

Comment:
```suggestion
		for i, name := range runnerinstall.Platforms {
```

I think you can simplify this for loop, and remove the  `i++`.